girder/girder_large_image/web_client/vue/components/CompositeChannels.vue

```diff
@@ -0,0 +1,189 @@
+        'color-picker': Chrome
+    },
+    data() {
+        return {
+            enabledChannels: this.channels,
+            colorPickerShown: undefined,
+            currentColorPickerRef: undefined,
+            compositeChannelInfo: {},
+        }
+    },
+    methods: {
+        toggleColorPicker(channel) {
+            this.colorPickerShown = channel
+            if (this.colorPickerShown === undefined) {
+                document.removeEventListener('click', this.documentClick);
+                // Only update style when picker is closed
+                this.updateStyle()
+            }
+            else {
+                this.currentColorPickerRef = document.getElementById(channel+'_picker')
+                document.addEventListener('click', this.documentClick);
+            }
+        },
+        documentClick(e) {
+            const picker = this.currentColorPickerRef;
+            if (picker && picker !== e.target && !picker.contains(e.target)) {
+                this.toggleColorPicker(undefined);
+            }
+        },
+        updateChannelColor(channel, swatch) {
+            this.compositeChannelInfo[channel].falseColor = swatch.hex;
+        },
+        updateChannelMin(event, channel) {
+            const newVal = event.target.value;
+            const newMinVal = parseFloat(newVal);
+            this.compositeChannelInfo[channel].min = newMinVal;
+            this.updateStyle();
+        },
+        updateChannelMax(event, channel) {
+            const newVal = event.target.valueAsNumber;
+            const newMaxVal = parseFloat(newVal);
+            this.compositeChannelInfo[channel].max = newMaxVal;
+            this.updateStyle();
+        },
+        updateActiveChannels() {
+            this.channels.forEach((channel) => {
+                this.compositeChannelInfo[channel].enabled = this.enabledChannels.includes(channel);
+            })
+            this.updateStyle();
+        },
+        updateStyle() {
+            const activeChannels = Object.values(
+                this.compositeChannelInfo
+            ).filter((channel) => channel.enabled);
+            const styleArray = []
+            activeChannels.forEach((channel) => {
+                const styleEntry = {
+                    frameDelta: channel.number,
+                };
+                if (channel.falseColor) {
+                    styleEntry['palette'] = channel.falseColor;
+                }
+                if (channel.min) {
+                    styleEntry['min'] = channel.min;
+                }
+                if (channel.max) {
+                    styleEntry['max'] = channel.max;
+                }
+                styleArray.push(styleEntry);
+            });
+            this.$emit('updateStyle', {bands: styleArray});
+        },
+    },
+    mounted() {
+        const usedColors = []
+        this.compositeChannelInfo = Object.fromEntries(this.channels
+        .sort((channel) => !Object.keys(CHANNEL_COLORS).includes(channel))
```

Comment:
`sort` takes a compare function that takes two parameters (`a` and `b`).  I'm not sure what only treating one value does, but the sort doesn't work the way we want.